sinks/kafka/kafka.go

```diff
@@ -0,0 +1,307 @@
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Shopify/sarama"
+	"github.com/gogo/protobuf/proto"
+	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/sinks"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
+)
+
+var _ sinks.MetricSink = &KafkaMetricSink{}
+var _ sinks.SpanSink = &KafkaSpanSink{}
+
+type KafkaMetricSink struct {
+	logger      *logrus.Entry
+	producer    sarama.AsyncProducer
+	statsd      *statsd.Client
+	checkTopic  string
+	eventTopic  string
+	metricTopic string
+	brokers     string
+	config      *sarama.Config
+}
+
+type KafkaSpanSink struct {
+	logger       *logrus.Entry
+	producer     sarama.AsyncProducer
+	statsd       *statsd.Client
+	topic        string
+	brokers      string
+	serializer   string
+	config       *sarama.Config
+	spansFlushed int64
+}
+
+// NewKafkaMetricSink creates a new Kafka Plugin.
+func NewKafkaMetricSink(logger *logrus.Logger, brokers string, checkTopic string, eventTopic string, metricTopic string, ackRequirement string, partitioner string, retries int, bufferBytes int, bufferMessages int, bufferDuration string, stats *statsd.Client) (*KafkaMetricSink, error) {
```

Comment:
Can we make the `logger` argument nil-safe? That is, if it's nil, we should create a new logger that discards all input.

Reply:
Sure! Is there a pattern for doing that aside from using a `StandardLogger`.

Reply:
`nullLogger := &logrus.Logger{Out: ioutil.Discard}` should do the trick.